ui/components/component-library/button-icon/button-icon.test.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable jest/require-top-level-describe */
+import { render } from '@testing-library/react';
+import React from 'react';
+import { BUTTON_ICON_SIZES } from './button-icon.constants';
+import { ButtonIcon } from './button-icon';
+
+describe('ButtonIcon', () => {
+  it('should render button element correctly', () => {
+    const { getByTestId, container } = render(
+      <ButtonIcon data-testid="button-icon" icon="add-square-filled" />,
+    );
+    expect(container.querySelector('button')).toBeDefined();
+    expect(getByTestId('button-icon')).toHaveClass('mm-button-icon');
+  });
+
+  it('should render anchor element correctly', () => {
+    const { getByTestId, container } = render(
+      <ButtonIcon as="a" data-testid="button-icon" />,
+    );
+    expect(getByTestId('button-icon')).toHaveClass('mm-button-icon');
+    const anchor = container.getElementsByTagName('a').length;
+    expect(anchor).toBe(1);
```

Comment:
You could also use `getByRole` here if you want to save some lines of code
```suggestion
    expect(getByRole('link')).toBeDefined();
```

Reply:
Ahh it will also need to include an href attribute

https://stackoverflow.com/questions/69725673/react-testing-library-cant-find-anchor-element-by-role

The below should pass
```
  it('should render anchor element correctly', () => {
    const { getByTestId, getByRole } = render(
      <ButtonIcon
        as="a"
        href="/metamask"
        data-testid="button-icon"
        icon="add-square-filled"
        ariaLabel="add"
      />,
    );
    expect(getByTestId('button-icon')).toHaveClass('mm-button-icon');
    expect(getByRole('link')).toBeDefined();
  });
```

Reply:
Sounds good!